components/feral-watchdog/cdp.go

```diff
@@ -0,0 +1,170 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"sync"
+	"time"
+
+	"go.uber.org/zap"
+)
+
+const (
+	// CDP configuration
+	CDP_CHECK_INTERVAL         = 5 * time.Second // Check CDP every 5 seconds
+	CDP_REQUEST_TIMEOUT        = 3 * time.Second
+	CDP_HANG_THRESHOLD         = 20 * time.Second
+	CDP_RESTART_HISTORY_SIZE   = 3 // Store the last 3 restarts
+	CDP_MAX_RESTARTS_WINDOW    = 5 * time.Minute
+	CDP_MAX_RESTARTS_THRESHOLD = 3 // 3 restarts within the window triggers reboot
+)
+
+// CDPMonitor monitors Chromium browser health via Chrome DevTools Protocol
+type CDPMonitor struct {
+	mu                 sync.Mutex
+	cdpConfig          *CDPConfig
+	client             *http.Client
+	logger             *zap.Logger
+	restartHistory     []time.Time
+	lastSuccessfulResp time.Time
+	commandHandler     *CommandHandler
+}
+
+// NewCDPMonitor creates a new CDP monitor instance
+func NewCDPMonitor(cdpConfig *CDPConfig, logger *zap.Logger, commandHandler *CommandHandler) *CDPMonitor {
+	return &CDPMonitor{
+		cdpConfig: cdpConfig,
+		client: &http.Client{
+			Timeout: CDP_REQUEST_TIMEOUT,
+		},
+		logger:             logger,
+		restartHistory:     make([]time.Time, 0, CDP_RESTART_HISTORY_SIZE),
+		lastSuccessfulResp: time.Time{},
+		commandHandler:     commandHandler,
+	}
+}
+
+// Start begins the CDP monitoring process
+func (m *CDPMonitor) Start(ctx context.Context) {
+	m.logger.Info("CDP: Starting Chromium CDP monitor",
+		zap.String("endpoint", m.cdpConfig.Endpoint),
+		zap.Duration("check_interval", CDP_CHECK_INTERVAL),
+		zap.Duration("hang_threshold", CDP_HANG_THRESHOLD))
+
+	ticker := time.NewTicker(CDP_CHECK_INTERVAL)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			m.logger.Info("CDP: Monitor shutting down")
+			return
+		case <-ticker.C:
+			if err := m.check(ctx); err != nil {
+				m.logger.Warn("CDP: Health check failed", zap.Error(err))
+			} else {
+				m.logger.Debug("CDP: Health check passed")
+			}
+		}
+	}
+}
+
+// check performs a single CDP health check
+func (m *CDPMonitor) check(ctx context.Context) error {
+	versionURL := fmt.Sprintf("%s/json/version", m.cdpConfig.Endpoint)
+
+	// Create context with timeout
+	ctx, cancel := context.WithTimeout(ctx, CDP_REQUEST_TIMEOUT)
```

Comment:
Small suggestion: we should use another name to avoid ambiguous context like `cdpCtx`